### PR TITLE
Spring framework wiring 101

### PR DESCRIPTION
**Pull request by barsifedron**

### CODE REVIEW

File: cqrs-spring-utils/build.gradle [hunk 1]
```diff
@@ -0,0 +1,6 @@
+dependencies {
+
+    implementation project(":bus-cqrs-api")
```
Comment (eskatos) [inline cqrs-spring-utils/build.gradle]: Should be `api`, the types from `:bus-cqrs-api` are exposed to consumers of this module right?
```suggestion
    api project(":bus-cqrs-api")
```

File: cqrs-spring-utils/build.gradle [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+
+    compile "org.springframework:spring-context:4.3.14.RELEASE"
```
Comment (eskatos) [inline cqrs-spring-utils/build.gradle]: Should be `api`, the types from `spring-context` are exposed to consumers of this module right?
```suggestion
    api "org.springframework:spring-context:4.3.14.RELEASE"
```

File: cqrs-spring-utils/src/main/java/com/barsifedron/candid/cqrs/spring/CommandHandlersRegistry.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Supplier;
+
+@Component
+@Scope(value = ConfigurableBeanFactory.SCOPE_SINGLETON)
+public class CommandHandlersRegistry {
+
+    private Map<Class<Command>, Supplier<CommandHandler>> map;
```
Comment (eskatos) [inline cqrs-spring-utils/src/main/java/com/barsifedron/candid/cqrs/spring/CommandHandlersRegistry.java]: :o: I would make that field `final` and initialized at declaration site, same comment for the two other registries

File: cqrs-spring-utils/src/main/java/com/barsifedron/candid/cqrs/spring/DomainEventHandlersRegistry.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        String[] names = applicationContext.getBeanNamesForType(DomainEventHandler.class);
+        for (String name : names) {
+            register(applicationContext, name);
+        }
+    }
+
+    private void register(ApplicationContext applicationContext, String name) {
+        Class<DomainEventHandler<?>> handlerClass = (Class<DomainEventHandler<?>>) applicationContext.getType(name);
+
+        System.out.println("handlerClass = " + handlerClass);
```
Comment (eskatos) [inline cqrs-spring-utils/src/main/java/com/barsifedron/candid/cqrs/spring/DomainEventHandlersRegistry.java]: :x: remove spurious printlns

File: cqrs-utils/src/main/java/com/barsifedron/candid/cqrs/query/middleware/ExceptionLoggingQueryBusMiddleware.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+    @Override
+    public <T> T dispatch(Query<T> query, QueryBus queryBus) {
+        try {
+
+            return queryBus.dispatch(query);
+
+        } catch (Throwable th) {
+
+            LOGGER.info(""
```
Comment (eskatos) [inline cqrs-utils/src/main/java/com/barsifedron/candid/cqrs/query/middleware/ExceptionLoggingQueryBusMiddleware.java]: ❓ I would have used the error logging level, was using info instead done on purpose? care to explain why? just curious :)

File: springboot/build.gradle [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+
+    implementation project(":bus-cqrs-api")
+    implementation project(":cqrs-utils")
+    implementation project(":cqrs-spring-utils")
+    compile "javax.inject:javax.inject:1"
```
Comment (eskatos) [inline springboot/build.gradle]: :x: don't use `compile` it is deprecated, prefer `implementation` or `api`

File: springboot/src/main/java/com/barsifedron/candid/cqrs/springboot/sample/writeside/command/PleaseDoSomethingCommand.java [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    @NotNull
+    public final Integer paramAboveZero;
+
+    @Min(1)
+    @Max(10)
+    @NotNull
+    public final Integer paramBetweenOneAndTen;
+
+    /**
+     * Not a big fan of lombok but there really is no shame in using it for this
```
Comment (eskatos) [inline springboot/src/main/java/com/barsifedron/candid/cqrs/springboot/sample/writeside/command/PleaseDoSomethingCommand.java]: :x: those comments are the only mention of lombok, my understanding is that it's not in use here and that the comments are irrelevant and should be removed

File: cqrs-utils/src/main/java/com/barsifedron/candid/cqrs/query/middleware/ExceptionLoggingQueryBusMiddleware.java (hunk 5 above)
Comment (barsifedron) [reply cqrs-utils/src/main/java/com/barsifedron/candid/cqrs/query/middleware/ExceptionLoggingQueryBusMiddleware.java]: just dumb :)

File: springboot/src/main/java/com/barsifedron/candid/cqrs/springboot/sample/writeside/command/PleaseDoSomethingCommand.java (hunk 7 above)
Comment (barsifedron) [reply springboot/src/main/java/com/barsifedron/candid/cqrs/springboot/sample/writeside/command/PleaseDoSomethingCommand.java]: fair point

File: cqrs-spring-utils/src/main/java/com/barsifedron/candid/cqrs/spring/DomainEventHandlersRegistry.java (hunk 4 above)
Comment (barsifedron) [reply cqrs-spring-utils/src/main/java/com/barsifedron/candid/cqrs/spring/DomainEventHandlersRegistry.java]: indeed

File: cqrs-spring-utils/src/main/java/com/barsifedron/candid/cqrs/spring/CommandHandlersRegistry.java (hunk 3 above)
Comment (barsifedron) [reply cqrs-spring-utils/src/main/java/com/barsifedron/candid/cqrs/spring/CommandHandlersRegistry.java]: deal

File: cqrs-utils/src/main/java/com/barsifedron/candid/cqrs/query/middleware/ExceptionLoggingQueryBusMiddleware.java (hunk 5 above)
Comment (eskatos) [reply cqrs-utils/src/main/java/com/barsifedron/candid/cqrs/query/middleware/ExceptionLoggingQueryBusMiddleware.java]: That's sounds like a very reasonable explanation I can relate to ;)

File: settings.gradle [hunk 8]
```diff
@@ -1,9 +1,12 @@
...
-    "bus-cqrs-example",
-    "cqrs-utils",
-    "cqrs-guice-utils",
-    "candid-domain"
+        "bus-cqrs-api",
+        "bus-cqrs-example",
+        "cqrs-utils",
+        "cqrs-guice-utils",
+        "candid-domain",
+        "springboot"
```
Comment (eskatos) [inline settings.gradle]: Given that the `:springboot` project is part of the overall build, you don't need the gradle wrapper files in `./springboot`, they can be removed. Moreover, you don't need the `./springboot/settings.gradle` script either.

**Pull request merged**